{"name": "登录127.0.0.1:62001", "status": "broken", "statusDetails": {"message": "FileNotFoundError: [Errno 2] No such file or directory: 'C:\\\\study\\\\pythonT\\\\pyAppium-master\\\\outputs\\\\picture\\\\我的_用户名20231019142019.png'", "trace": "self = <pageViwe.loginPage.LoginPage object at 0x0000023A55E0C4F0>\ndoc = '我的_用户名', loc = ('id', 'com.xkw.client:id/mine_username')\n\n    def find_element(self, *loc, doc=\"\") -> WebElement:\n        \"\"\"\n        查找单个元素方法\n        :type doc: 定位元素界面位置：例如 首页\n        :type loc: 元素定位\n        \"\"\"\n        try:\n            logging.info(\"{} 页面开始查找元素{}\".format(doc, loc))\n>           WebDriverWait(self.driver, 20, 0.5).until(EC.visibility_of_element_located(loc))\n\npageViwe\\basePage.py:34: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"391aa67e-3d21-4eee-b642-10d1623d7630\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x0000023A55B19040>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       NoSuchElementError: An element could not be located on the page using the given search parameters.\nE           at AndroidUiautomator2Driver.findElOrEls (C:\\Users\\YM520\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\appium-android-driver\\lib\\commands\\find.js:75:11)\nE           at processTicksAndRejections (node:internal/process/task_queues:95:5)\nE           at AndroidUiautomator2Driver.findElOrElsWithProcessing (C:\\Users\\YM520\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\appium-base-driver\\lib\\basedriver\\commands\\find.js:33:12)\nE           at AndroidUiautomator2Driver.findElement (C:\\Users\\YM520\\AppData\\Roaming\\npm\\node_modules\\appium\\node_modules\\appium-base-driver\\lib\\basedriver\\commands\\find.js:53:10)\n\nC:\\python3.8\\lib\\site-packages\\selenium\\webdriver\\support\\wait.py:95: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <testcase.test_login.TestLogin object at 0x0000023A55DEB730>\ncommon_driver = <pageViwe.loginPage.LoginPage object at 0x0000023A55E0C4F0>\ndevice = {'device': '127.0.0.1:62001', 'platform_version': '7.1.2', 'server_port': 4723, 'system_port': 8200}\n\n    @allure.story(\"正常登录\")\n    @allure.description(\"用户正常登录\")  # 用例的描述\n    @allure.severity(allure.severity_level.BLOCKER)\n    def test_login(self, common_driver, device):\n        allure.dynamic.title(\"登录\" + device[\"device\"])\n        common_driver.login('XXX', 'XXX')\n>       common_driver.result_assert(common_driver.get_username(), \"xkw_03314932\", '用户正常登录')\n\ntestcase\\test_login.py:27: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npageViwe\\loginPage.py:59: in get_username\n    return self.get_text(*self.login_btn, doc=\"我的_用户名\")\npageViwe\\basePage.py:116: in get_text\n    ele = self.find_element(*loc, doc=doc)\npageViwe\\basePage.py:40: in find_element\n    screen_name = self.get_screenshot(doc)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <pageViwe.loginPage.LoginPage object at 0x0000023A55E0C4F0>\ndoc = '我的_用户名'\n\n    def get_screenshot(self, doc):\n        \"\"\"\n        截图\n        \"\"\"\n        logging.info(\"开始进行截图..\")\n        now = datetime.datetime.now().strftime('%Y%m%d%H%M%S')\n        pic_name = picturePath + r\"\\{}\".format(doc) + now + '.png'\n        self.driver.get_screenshot_as_file(pic_name)\n        logging.info(\"截图成功！图片名称为:{}\".format(pic_name))\n>       with open(pic_name, mode='rb') as f:\nE       FileNotFoundError: [Errno 2] No such file or directory: 'C:\\\\study\\\\pythonT\\\\pyAppium-master\\\\outputs\\\\picture\\\\我的_用户名20231019142019.png'\n\npageViwe\\basePage.py:212: FileNotFoundError"}, "description": "用户正常登录", "steps": [{"name": "我的页面，点击切换到我的页面", "status": "passed", "start": 1697696389179, "stop": 1697696390267}, {"name": "我的页面，点击登录", "status": "passed", "start": 1697696390267, "stop": 1697696391737}, {"name": "我的_登录页面_切换到用户密码登录", "status": "passed", "start": 1697696391737, "stop": 1697696393299}, {"name": "我的_账号密码登录页面，输入账号:XXX", "status": "passed", "start": 1697696393299, "stop": 1697696396260}, {"name": "我的_账号密码登录页面，输入密码:XXX", "status": "passed", "start": 1697696396260, "stop": 1697696397968}, {"name": "我的_账号密码登录页面_点击提交登录", "status": "passed", "start": 1697696397968, "stop": 1697696398608}, {"name": "获取用户名", "status": "broken", "statusDetails": {"message": "FileNotFoundError: [Errno 2] No such file or directory: 'C:\\\\study\\\\pythonT\\\\pyAppium-master\\\\outputs\\\\picture\\\\我的_用户名20231019142019.png'\n", "trace": "  File \"C:\\study\\pythonT\\pyAppium-master\\pageViwe\\loginPage.py\", line 59, in get_username\n    return self.get_text(*self.login_btn, doc=\"我的_用户名\")\n  File \"C:\\study\\pythonT\\pyAppium-master\\pageViwe\\basePage.py\", line 116, in get_text\n    ele = self.find_element(*loc, doc=doc)\n  File \"C:\\study\\pythonT\\pyAppium-master\\pageViwe\\basePage.py\", line 40, in find_element\n    screen_name = self.get_screenshot(doc)\n  File \"C:\\study\\pythonT\\pyAppium-master\\pageViwe\\basePage.py\", line 212, in get_screenshot\n    with open(pic_name, mode='rb') as f:\n"}, "start": 1697696398609, "stop": 1697696420295}], "attachments": [{"name": "log", "source": "11dc26b2-b98f-45be-9ed1-f416cd512abb-attachment.txt", "type": "text/plain"}], "start": 1697696389179, "stop": 1697696420301, "uuid": "40a871b7-3e2a-4b62-a4da-01a94627d3d1", "historyId": "99fa6618033e0c4c42d1151140fc593a", "testCaseId": "99fa6618033e0c4c42d1151140fc593a", "fullName": "309_testcase.test_login.TestLogin#test_login", "labels": [{"name": "story", "value": "正常登录"}, {"name": "severity", "value": "blocker"}, {"name": "feature", "value": "登录"}, {"name": "parentSuite", "value": "309_testcase"}, {"name": "suite", "value": "test_login"}, {"name": "subSuite", "value": "TestLogin"}, {"name": "host", "value": "DESKTOP-CHVCFJK"}, {"name": "thread", "value": "1040-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "309_testcase.test_login"}]}